packages/graph/hash_graph/lib/graph/src/datastore/mod.rs

```diff
@@ -146,19 +148,43 @@ impl fmt::Display for DatabaseConnectionInfo {
 /// Describes the API of a Datastore implementation
 #[async_trait]
 trait Datastore {
-    async fn create_data_type() -> Result<(), DatastoreError>;
-
-    async fn get_data_type() -> Result<(), DatastoreError>;
+    /// Creates a new [`DataType`].
+    ///
+    /// # Errors:
+    ///
+    /// If the account referred to by `created_by` does not exist.
+    async fn create_data_type(
+        &self,
+        data_type: DataType,
+        created_by: AccountId,
+    ) -> Result<Qualified<DataType>, DatastoreError>;
+
+    /// Get an existing [`DataType`] by an [`Identifier`].
+    ///
+    /// # Errors
+    ///
+    /// If the [`DataType`] doesn't exist.
+    async fn get_data_type(&self, id: Identifier) -> Result<Qualified<DataType>, DatastoreError>;
 
-    async fn get_data_many() -> Result<(), DatastoreError>;
+    async fn get_data_type_many() -> Result<(), DatastoreError>;
 
-    async fn update_data_type() -> Result<(), DatastoreError>;
+    /// Update the definition of an exisitng [`DataType`].
+    ///
+    /// # Errors
+    ///
+    /// If the [`DataType`] doesn't exist.
```

Comment:
```suggestion
    /// - [`DatastoreError`], if the [`DataType`] doesn't exist.
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/734/commits/4b14fd9e283d0d74de3caaf3a34e165e7c318c09